julia/core.py

```diff
@@ -271,42 +280,66 @@ def __init__(self, init_julia=True, jl_init_path=None):
         self.api.jl_exception_occurred.restype = void_p
         self.api.jl_typeof_str.argtypes = [void_p]
         self.api.jl_typeof_str.restype = char_p
-        self.api.jl_call1.restype = void_p
+        self.api.jl_call2.argtypes = [void_p, void_p, void_p]
+        self.api.jl_call2.restype = void_p
         self.api.jl_get_field.restype = void_p
         self.api.jl_typename_str.restype = char_p
         self.api.jl_typeof_str.restype = char_p
         self.api.jl_unbox_voidpointer.restype = py_object
-        self.api.jl_bytestring_ptr.restype = char_p
 
-        if init_julia:
-            try:
-                self.call('using PyCall')
-            except:
-                raise JuliaError("Julia does not have package PyCall.\n"
-                    "Install PyCall by running the following line:\n"
-                    """\tjulia -e 'Pkg.add("PyCall"); Pkg.update()'\n""")
-            try:
-                self.call('pyinitialize(C_NULL)')
-            except:
-                raise JuliaError("Failed to initialize PyCall package")
+        self.api.jl_exception_clear.restype = None
+        self.api.jl_stderr_obj.argtypes = []
+        self.api.jl_stderr_obj.restype = void_p
+        self.api.jl_stderr_stream.argtypes = []
+        self.api.jl_stderr_stream.restype = void_p
+        self.api.jl_show.restype = None
+        self.api.jl_show.argtypes = [void_p, void_p]
+        self.api.jl_printf.restype = ctypes.c_int
+        self.api.jl_exception_clear()
 
+        if init_julia:
+            # Tell PyCall it's ok to use symbols from the current process
+            os.environ["PYCALL_CURRENT_PROCESS"]="1"
+            os.environ["PYCALL_PYTHON_EXE"] = sys.executable
+            os.environ["PYCALL_JULIA_HOME"] = JULIA_HOME
+            self._call(u"eval(Base,:(JULIA_HOME=\"/Users/kfischer/Projects/pyjulia/fake-julia\"))")
+            # Replace the cache directory with a private one. PyCall needs a different
+            # configuration and so do any packages that depend on it. Ideally, we could
+            # detect packages that depend on PyCall and only use LOAD_CACHE_PATH for them
+            # but that would be significantly more complicated and brittle, and may not
+            # be worth it.
+            self._call(u"empty!(Base.LOAD_CACHE_PATH)")
+            self._call(u"push!(Base.LOAD_CACHE_PATH, abspath(Pkg.Dir._pkgroot()," +
```

Comment:
You mean `_pkgroot`? Unfortunately that's how to determine where this directory is. If you want to advocate for exporting that method from Base, be my guest.